crates/swc_plugin_runner/src/imported_fn/comments.rs

```diff
@@ -0,0 +1,70 @@
+use swc_common::{comments::Comments, plugin::Serialized, BytePos};
+use swc_plugin_comments::{CommentsVecPtr, HostCommentsStorage, COMMENTS};
+
+use crate::{host_environment::HostEnvironment, memory_interop::write_into_memory_view};
+
+/// Ask to get leading_comments from currently scoped comments held by
+/// HostCommentsStorage.
+///
+/// Returns 1 if operation success with Some(Vec<Comments>), 0 otherwise.
+/// Allocated results should be read through CommentsPtr.
+pub fn get_leading_comments_proxy(
+    env: &HostEnvironment,
+    byte_pos: u32,
+    allocated_ret_ptr: i32,
+) -> i32 {
+    if let Some(memory) = env.memory_ref() {
+        if let Some(alloc_guest_memory) = env.alloc_guest_memory_ref() {
+            if COMMENTS.is_set() {
+                return COMMENTS.with(|storage: HostCommentsStorage| {
+                    if let Some(comments) = storage.inner {
+                        let leading_comments = comments.get_leading(BytePos(byte_pos));
+                        if let Some(leading_comments) = leading_comments {
+                            let serialized_leading_comments_vec_bytes =
+                                Serialized::serialize(&leading_comments)
+                                    .expect("Should be serializable");
+
+                            let serialized_bytes_len =
+                                serialized_leading_comments_vec_bytes.as_ref().len();
+
+                            let (allocated_ptr, allocated_ptr_len) = write_into_memory_view(
```

Comment:
Since `get_leading` returns `Option<Vec<Comment>>`, plugin cannot pre-allocate size of the return value similar to others. Instead, it preallocates `CommentsVecPtr(ptr, len)` to let host writes ptr to the vec for the return value. In the host, it allocates return value's memory directly via hostEnvironment::alloc, then writes its ptr with length to guest's commentsVecPtr.